src/Loggers/OwnFilteringChannel.cpp

```diff
@@ -0,0 +1,47 @@
+#include "OwnFilteringChannel.h"
+#include <Poco/RegularExpression.h>
+
+
+namespace DB
+{
+
+void OwnFilteringChannel::log(const Poco::Message & msg)
+{
+    std::string formatted_text;
+
+    // Apply formatting to the text
+    if (pFormatter)
+    {
+        pFormatter->formatExtended(ExtendedLogMessage::getFrom(msg), formatted_text);
```

Comment:
I am afraid this can cause some serious issues as we can log a lot of messages. If we format the log messages twice (once here, once before we outputting them), then the logging might be slowed down significantly. Obviously this is only speculation and kind of a worst-case scenario, but it would be nice to see the impact of this in some real examples.

Reply:
That's what I was concerned about. The coloring and formatting are done in one function. I'm not aware of any way we can decouple them into separate functions. Since we don't want two passes, I'm not seeing a perfect solution. I think the next best alternative would be one of the following.

- Skip the filtering process if no patterns are specified. This would isolate the slowdown to only cases where this feature is used. We could add a note about the slowdown in the docs.
(I just implemented this way in my most recent commit, but tests are still failing)

- Only one pass in all cases, but incorrect behavior for colored output and less clean code - I could instead add duplicate filtering channels after each formatter that already exists in the split channel. The color characters would also be passed to the filter since we want the formatted text, misleading the user. We could put a note in the docs to warn about this. (Or maybe we could try filtering the color characters out in that case, but that'll be pretty slow again)

I think at minimum, we should do the first bullet, but I'm not sure about the second one. @antaljanosbenjamin What do you think? Would this be good enough?

Reply:
The first one is a must have as you said and implemented, nicely done. I think the second one is very very questionable. Let's go with this as is right now and see how it turns out. Maybe add a note about this in the docs.